redsky/event_streams.py

```diff
@@ -0,0 +1,711 @@
+                                       zzip(self.output_info, outputs)})
+            else:
+                new_event.update(data=outputs['data'])
+            self.i += 1
+            return new_event
+
+    def refresh_event(self, event):
+        """Issue a new event
+
+        Parameters
+        ----------
+        event: tuple, dict, or other
+            The event document
+        Returns
+        -------
+        new_event: dict
+            A new event with the same core data
+        """
+        if not self.event_failed:
+            if self.run_start_uid is None:
+                raise RuntimeError("Received Event before RunStart.")
+            if isinstance(event, Exception):
+                return self.stop(event)
+
+            new_event = dict(event)
+            new_event.update(dict(uid=str(uuid.uuid4()),
+                                  time=time.time(),
+                                  timestamps={},
+                                  seq_num=self.i))
+
+            self.i += 1
+            return new_event
+
+
+class map(EventStream):
+    """Apply a function onto every event in the stream"""
+    def __init__(self, func, child, *,
+                 output_info=None, input_info=None,
+                 **kwargs):
+        """Initialize the node
+
+        Parameters
+        ----------
+        func: callable
+            The function to map on the event data
+        child: EventStream instance
+            The source of the data
+        input_info: dict
+            describe the incoming streams
+        output_info: list of tuples
+            describe the resulting stream
+        """
+        self.func = func
+        self.kwargs = kwargs
+
+        EventStream.__init__(self, child, output_info=output_info,
+                             input_info=input_info, **kwargs)
+        self.generate_provenance(func)
+
+    def event(self, docs):
+        try:
+            # we need to expose the raw event data
+            res = self.event_guts(docs)
+            result = self.func(res, **self.kwargs)
+            # Now we must massage the raw return into a new event
+            result = self.issue_event(result)
+        except Exception as e:
+            result = self.issue_event(e)
+        return super().event(result)
+
+
+class filter(EventStream):
+    """Only pass through events that satisfy the predicate"""
+    def __init__(self, predicate, child, *, input_info,
+                 full_event=False, **kwargs):
+        """Initialize the node
+
+        Parameters
+        ----------
+        predicate: callable
+            The function which returns True if the event is to propagate
+            further in the pipeline
+        child: EventStream instance
+            The source of the data
+        input_info: dict
+            describe the incoming streams
+        full_event: bool, optional
+            If True expose the full event dict to the predicate, if False
```

Comment:
yeah, I know, and it was a good idea to remove it since it added too much complication. I was more asking if `raw` still made more sense as a keyword than `full_event`. I personally prefer it as I find `full_event` misleading to me (when `False` it's not a partial event, just an event innard). But this is really a small issue.
no worries about the master push.